pkg/core/basic_cluster.go

```diff
@@ -97,7 +99,29 @@ type RegionSetInformer interface {
 	GetAdjacentRegions(region *RegionInfo) (*RegionInfo, *RegionInfo)
 	ScanRegions(startKey, endKey []byte, limit int) []*RegionInfo
 	GetRegionByKey(regionKey []byte) *RegionInfo
-	BatchScanRegions(keyRanges *KeyRanges, limit int) []*RegionInfo
+	BatchScanRegions(keyRanges *KeyRanges, opts ...BatchScanRegionsOptionFunc) ([]*RegionInfo, error)
```

Comment:
This is an internal method, maybe bool is enough.

Reply:
You didn't say that before, did you? 😭
https://github.com/tikv/pd/pull/8375#discussion_r1670120502

Reply:
Both are ok to me.